README.md

```diff
@@ -13,121 +13,274 @@ and the Flutter guide for
 
 # Solid Auth
 
-Solid Auth is an implementation of [Solid-OIDC flow](https://solid.github.io/solid-oidc/) which can be used to authenticate a client application to a Solid POD. Solid OIDC is built on top of OpenID Connect 1.0.
+[![pub package](https://img.shields.io/pub/v/solid_auth.svg)](https://pub.dev/packages/solid_auth)
 
-The authentication process works with both Android and Web based client applications. The package can also be used to create DPoP proof tokens for accessing private data inside PODs after the authentication.
+A Flutter library for authenticating with [Solid pods](https://solidproject.org/) using OpenID Connect, implementing the [Solid-OIDC specification](https://solid.github.io/solid-oidc/).
 
-This package includes the source code of two other packages, [openid_client](https://pub.dev/packages/openid_client) and [dart_jsonwebtoken](https://pub.dev/packages/dart_jsonwebtoken), with slight modifications done to those package files in order to be compatible with Solid-OIDC flow.
+This library provides a simple, reactive interface for Solid authentication that handles the complexity of OIDC flows, token management, WebID discovery, and DPoP (Demonstration of Proof-of-Possession) tokens required by Solid servers.
 
-## Features
+Built on the robust foundation of [Bdaya-Dev/oidc](https://pub.dev/packages/oidc), this package focuses specifically on Solid pod authentication while leveraging excellent, well-maintained OpenID Connect functionality.
 
-* Authenticate a client application to a Solid POD
-* Create DPoP tokens for accessing data inside a POD
-* Access public profile data of a POD using its WebID
+This package includes an embedded copy of the [dart_jsonwebtoken](https://pub.dev/packages/dart_jsonwebtoken) package with modifications for Solid-OIDC compatibility.
+
+## ✨ Features
+
+- **🔐 Complete Solid Authentication**: Full implementation of Solid-OIDC specification
+- **📱 Cross-Platform**: Works on web, mobile (iOS/Android), and desktop (macOS)
+- **🔄 Reactive State Management**: Use `ValueListenable` to reactively update UI based on authentication status
+- **💾 Automatic Session Restoration**: Persists authentication across app restarts
+- **🛡️ DPoP Token Support**: Handles security tokens required by Solid servers
+- **🌐 WebID Discovery**: Automatically discovers identity providers from WebIDs
+- **🔒 Secure Token Storage**: Uses platform-appropriate secure storage mechanisms
 
 <!-- ## Getting started
 
 TODO: List prerequisites and provide or point to information on how to
 start using the package. -->
 
-## Usage
+## 🚀 Quick Start
+
+### 1. Add to pubspec.yaml
+
+```sh
+dart pub add solid_auth
+```
+
+### 2. Create Your Client Profile
```

Comment:
Do you think we should discuss the options available in general for OIDC client identification (like preregistering and dynamic registration) and why only the client profile approach is suitable for flutter solid apps? Maybe as a separate markdown document?